lib/gollum/views/commit.rb

```diff
@@ -0,0 +1,111 @@
+module Precious
+  module Views
+    class Commit < Layout
+      include HasPage
+
+      attr_reader :version
+
+      def title
+        "Changes in #{@version[1..7]}: #{message}"
+      end
+
+      def author
+        @commit.author.name
+      end
+
+      def authored_date
+        @commit.authored_date
+      end
+
+      def message
+        @commit.message
+      end
+
+      def files
+        puts @diff
+        files = @diff.split(%r{^diff --git a/.+ b/.+$}).reject(&:empty?)
+        files.map do |diff|
+          matched = diff.match(%r{(?<=^--- a/).+$})
+          matched = diff.match(%r{(?<=^\+\+\+ b/).+$}) if matched.nil?
+          {
+            path: matched[0],
+            lines: lines(diff)
+          }
+        end
+      end
+
+      private
+
+      def lines(diff)
```

Comment:
@dometto Do we want methods like this in `gollum` or preferably in `gollum-lib`?

Reply:
Fine to have them in `gollum` since we already have them [here](https://github.com/gollum/gollum/blob/master/lib/gollum/views/compare.rb). My main comment is thus that a) either these methods which are common to the `Commit` and `Compare` view should be in their own Module (like [here](https://github.com/gollum/gollum/blob/master/lib/gollum/views/has_user_icons.rb) for example), or the `Commit` view should just be a subclass of the `Compare` view.

Reply:
👍 Yes, let's try to avoid code duplication.

Reply:
Thank you for the suggestion. Now I learned `require_relative` and class inheritance in Ruby :)